parser.rb

```diff
@@ -46,29 +47,31 @@ def parse_ledger(file_path)
       end
 
       if transaction_match
-        action = line[ACCOUNT_ACTION_RGX]
-        amount = action ? action.scan(ACCOUNT_AMOUNT_RGX).join : nil
-        currency = action ? action[ACCOUNT_CURRENCY_RGX] : "USD"
+        if action = line[ACCOUNT_ACTION_RGX]
+          amount = action.scan(ACCOUNT_AMOUNT_RGX).join
+          currency = action[ACCOUNT_CURRENCY_RGX]
+        end
+
+        currency = 'USD' if currency.eql?('$')
```

Comment:
it's a good practice to keep these kind of strings like `USD` and `$` into constants, so it would end up with something like

```
        currency = USD if currency.eql?(DOLLAR_SIGN)
```